### PR TITLE
Negative invoice

### PR DESCRIPTION
**Pull request by yurypat**

### CODE REVIEW

File: src/Mews.Fiscalization.Greece/Model/PaymentBase.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using Mews.Fiscalization.Greece.Model.Types;
+using System;
+
+namespace Mews.Fiscalization.Greece.Model
+{
+    public abstract class PaymentBase
```
Comment (KaliCZ) [inline src/Mews.Fiscalization.Greece/Model/PaymentBase.cs:6]: Just `payment` and rename `Payment` to `PositivePayment`

File: src/Mews.Fiscalization.Greece/Model/RevenueItem.cs [hunk 2]
```diff
@@ -1,49 +1,18 @@
 ﻿using Mews.Fiscalization.Greece.Model.Types;
-using System;
 using System.Collections.Generic;
-using System.Linq;
 
 namespace Mews.Fiscalization.Greece.Model
 {
-    public class RevenueItem
+    public class RevenueItem : RevenueItemBase
```
Comment (KaliCZ) [inline src/Mews.Fiscalization.Greece/Model/RevenueItem.cs:6]: Same here. Let's name this `PositiveRevenue` and the base class just `Revenue`

File: src/Mews.Fiscalization.Greece/Model/ItemIncomeClassificationBase.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿using Mews.Fiscalization.Greece.Model.Types;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Mews.Fiscalization.Greece.Model
+{
+    public abstract class ItemIncomeClassificationBase
```
Comment (KaliCZ) [inline src/Mews.Fiscalization.Greece/Model/ItemIncomeClassificationBase.cs:8]: We could actually keep this as it was and just introduce `Amount` as a base class for `PositiveAmount` and `NegativeAmount`.

That way we won't need `ItemIncomeClassification` and `NegativeItemIncomeClassification`

File: src/Mews.Fiscalization.Greece/Model/InvoiceHeader.cs [hunk 4]
```diff
@@ -3,28 +3,11 @@
 
 namespace Mews.Fiscalization.Greece.Model
 {
-    public class InvoiceHeader
+    public class InvoiceHeader : InvoiceHeaderBase
```
Comment (KaliCZ) [inline src/Mews.Fiscalization.Greece/Model/InvoiceHeader.cs:6]: `InvoiceHeader` => `PositiveInvoiceHeader`
`InvoiceHeaderBase` => `InvoiceHeader`

File: src/Mews.Fiscalization.Greece/Model/InvoiceBase.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
+﻿using Mews.Fiscalization.Greece.Model.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Mews.Fiscalization.Greece.Model
+{
+    public abstract class InvoiceBase
```
Comment (KaliCZ) [inline src/Mews.Fiscalization.Greece/Model/InvoiceBase.cs:8]: Same here
`Invoice` => `PositiveInvoice`
`InvoiceBase` => `Invoice`

**Review by KaliCZ — APPROVED** [review]

File: src/Mews.Fiscalization.Greece/Model/PaymentBase.cs (hunk 1 above)
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/PaymentBase.cs:6]: Will change, thanks

File: src/Mews.Fiscalization.Greece/Model/RevenueItem.cs (hunk 2 above)
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/RevenueItem.cs:6]: Will change

File: src/Mews.Fiscalization.Greece/Model/ItemIncomeClassificationBase.cs (hunk 3 above)
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/ItemIncomeClassificationBase.cs:8]: Will change

File: src/Mews.Fiscalization.Greece/Model/InvoiceHeader.cs (hunk 4 above)
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/InvoiceHeader.cs:6]: Will change

File: src/Mews.Fiscalization.Greece/Model/InvoiceBase.cs (hunk 5 above)
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/InvoiceBase.cs:8]: Will change

**Pull request merged**